### ISSUE TITLE
Test failure: `OAuth2ClientCredentialsGrantTest.testConcurrent()`

### ISSUE THREAD

**Issue by trustin** [line/armeria#3687]
```
OAuth2ClientCredentialsGrantTest > testConcurrent() FAILED
    java.util.concurrent.CompletionException: java.lang.IllegalStateException: An aggregated message does not have headers.
        at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:412)
        at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2108)
        at com.linecorp.armeria.common.util.EventLoopCheckingFuture.join(EventLoopCheckingFuture.java:88)
        at com.linecorp.armeria.client.auth.oauth2.OAuth2ClientCredentialsGrantTest.validateResponses(OAuth2ClientCredentialsGrantTest.java:239)
        at com.linecorp.armeria.client.auth.oauth2.OAuth2ClientCredentialsGrantTest.testConcurrent(OAuth2ClientCredentialsGrantTest.java:215)

        Caused by:
        java.lang.IllegalStateException: An aggregated message does not have headers.
            at com.linecorp.armeria.internal.shaded.guava.base.Preconditions.checkState(Preconditions.java:510)
            at com.linecorp.armeria.common.HttpMessageAggregator.lambda$aggregateResponse$2(HttpMessageAggregator.java:120)
            at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
            at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
            at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2137)
            at com.linecorp.armeria.common.stream.DefaultStreamMessage.tryCollect(DefaultStreamMessage.java:637)
            at com.linecorp.armeria.common.stream.DefaultStreamMessage.lambda$tryCollect$6(DefaultStreamMessage.java:621)
            at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
            at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
            at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
            at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
            at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
            at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
            at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by mauhiz** [comment]
Reproduced in the wild :)

**Comment by ikhoon** [comment]
This was fixed by #3676.
